src/main/java/com/sarapis/orservice/controller/ServiceAreaController.java

```diff
@@ -0,0 +1,56 @@
+package com.sarapis.orservice.controller;
+
+import com.sarapis.orservice.dto.ServiceAreaDTO;
+import com.sarapis.orservice.service.ServiceAreaService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/api/service_areas")
+public class ServiceAreaController {
+
+    private final ServiceAreaService serviceAreaService;
+
+    @Autowired
+    public ServiceAreaController(ServiceAreaService serviceAreaService) { //stack overflow the dependency injection paste error and then @autowire
+        this.serviceAreaService = serviceAreaService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<ServiceAreaDTO>> getAllServiceAreas() {
+        List<ServiceAreaDTO> serviceAreaDTOs = this.serviceAreaService.getAllServiceAreas();
```

Comment:
You don't need to use `this` when calling the `serviceAreaService` as you already set the attribute in the Constructor of the class. Please change it to `serviceAreaService` instead

Reply:
Same comment for every other usage of `serviceAreaService`